it/uk/gov/hmrc/agentsubscriptionfrontend/controllers/StartControllerISpec.scala

```diff
@@ -212,21 +212,29 @@ class StartControllerISpec extends BaseISpec {
       redirectLocation(result).head should include(routes.CheckAgencyController.showCheckBusinessType().url)
     }
 
-    "delete the persisted KnownFactsResult if given a valid KnownFactsResult ID" in new ValidKnownFactsCached {
+    "delete the persisted KnownFactsResult if given a valid KnownFactsResult ID" in new ValidKnownFactsCached with UnsubscribedAgentStub {
       await(controller.returnAfterGGCredsCreated(id = Some(persistedId))(FakeRequest()))
 
-      await(repo.findKnownFactsResult(persistedId)) shouldBe None
+      await(repo.findChainedSessionDetails(persistedId)) shouldBe None
     }
 
-    "repopulate the KnownFacts session store with the persisted KnownFactsResult, if given a valid KnownFactsResult ID" in new ValidKnownFactsCached {
+    "place the known facts back in the session store, if given a valid ChainedSessionDetails ID" in new ValidKnownFactsCached with UnsubscribedAgentStub {
```

Comment:
its not just the known facts, you're storing the mapping boolean as well

Reply:
Yep, the next test covers just that